src/test/clojure/xtdb/aws/minio_test.clj

```diff
@@ -156,6 +159,33 @@
         ;; Ensure some files written to buffer-pool
         (t/is (seq (.listAllObjects buffer-pool)))))))
 
+(t/deftest ^:minio network-counter-test
+  (util/with-tmp-dirs #{local-disk-cache}
+    (util/with-open [node (start-node local-disk-cache (random-uuid))]
+      (let [registry ^CompositeMeterRegistry (tu/component node :xtdb.metrics/registry)
+            ^RemoteBufferPool buffer-pool (bp-test/fetch-buffer-pool-from-node node)]
+        ;; Submit some documents to the node
+        (.add registry (SimpleMeterRegistry.))
+        (t/is (= true
+                 (:committed? (xt/execute-tx node [[:put-docs :bar {:xt/id "bar1"}]
+                                                   [:put-docs :bar {:xt/id "bar2"}]
+                                                   [:put-docs :bar {:xt/id "bar3"}]]))))
+
+        ;; Ensure finish-block! works
+
+        (t/is (nil? (tu/finish-block! node)))
+
+        ;; Ensure can query back out results
+        (t/is (= [{:e "bar2"} {:e "bar1"} {:e "bar3"}]
+                 (xt/q node '(from :bar [{:xt/id e}]))))
+
+        ;; Ensure some files written to buffer-pool
+        (t/is (seq (.listAllObjects buffer-pool)))
+
+        ;; Ensure we have some network read/write
+        (t/is (< 100.0 (.count ^Counter (.counter (.find registry "buffer-pool.network.write")))))
+        (t/is (< 100.0 (.count ^Counter (.counter (.find registry "buffer-pool.network.read")))))))))
+
```

Comment:
Can we try to write this test purely on the buffer-pool and object-storage? Put some buffers into the object-storage, get them via the buffer pool, see the that network counters increase. Getting them a second time via the buffer pool doesn't increase the network counters.

For the write side you need to go through the buffer pool for putting.  Try to test the different paths.

If you need inspiration have a look at `buffer_pool_test.clj`

Reply:
sorry @tggreene - how was this one resolved?

Reply:
This test replaces a previous one which just looked for "some" read and write taking place of an arbitrary size (< 100 x).